### PR TITLE
Enable automatic labeling of PRs via Actions

### PR DESCRIPTION
**Pull request by nrb**
This PR introduces 2 ways of labeling PRs:

* Automatically via glob patterns with the [labeler](https://github.com/actions/labeler/) Action
* Allows users to apply labels via the `/kind` or `/area` commands via the [prow-github](https://github.com/marketplace/actions/prow-github-actions) Action

It also adds a checklist that contributors should fill out upon PR creation, and updates the code standards docs to reference it.

### CODE REVIEW

File: .github/workflows/prow-action.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    types: [created]
+
+jobs:
+  execute:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: jpmcb/prow-github-actions@v1
+        with:
+          # Only support /kind command for now.
+          # TODO: before allowing the /lgtm command, see if we can block merging if changelog labels are missing.
```
Comment (nrb) [inline .github/workflows/prow-action.yml:15]: I'm wondering if we can use the `/lgtm` command, but make it work *only if* there's either a `has-changelog` (which should be set with the action on this PR) or `changelog-not-required` label on the PR.

I think this way, we could retire the confusing changelog CI check that marks a perfectly-fine PR as failing.

Comment (carlisia) [reply .github/workflows/prow-action.yml:15]: /lgtm 😆

**Review by carlisia — DISMISSED** [review]
Booyah, love this! 🎉 

Approving but I am +1 on adding the /lgtm if you want to do that.

**Comment by nrb** [comment]
> Approving but I am +1 on adding the /lgtm if you want to do that.

I think I'll leave it open and address that later. This PR changes a lot already, and I wanna make sure it works :)

**Review by carlisia — APPROVED** [review]

**Review by zubron — APPROVED** [review]
Ooh, this is exciting - thanks! :tada:

**Pull request merged**